flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/CommittableToTableChangeConverter.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.WriteResult;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Internal
+public class CommittableToTableChangeConverter
+    extends ProcessFunction<CommittableMessage<IcebergCommittable>, TableChange> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(CommittableToTableChangeConverter.class);
+
+  private final FileIO io;
+  private final String tableName;
+  private final Map<Integer, PartitionSpec> specs;
+  private transient String flinkJobId;
+
+  public CommittableToTableChangeConverter(
+      FileIO fileIO, String tableName, Map<Integer, PartitionSpec> specs) {
+    Preconditions.checkNotNull(fileIO, "FileIO should not be null");
+    Preconditions.checkNotNull(tableName, "TableName should not be null");
+    Preconditions.checkNotNull(specs, "Specs should not be null");
+    this.io = fileIO;
+    this.tableName = tableName;
+    this.specs = specs;
+  }
+
+  @Override
+  public void open(OpenContext openContext) throws Exception {
+    super.open(openContext);
+    Preconditions.checkState(
+        getRuntimeContext().getTaskInfo().getNumberOfParallelSubtasks() == 1,
+        "CommittableToTableChangeConverter must run with parallelism 1, current parallelism: %s",
+        getRuntimeContext().getTaskInfo().getNumberOfParallelSubtasks());
+
+    this.flinkJobId = getRuntimeContext().getJobId().toString();
+  }
+
+  @Override
+  public void processElement(
+      CommittableMessage<IcebergCommittable> value, Context ctx, Collector<TableChange> out)
+      throws Exception {
+    if (value instanceof CommittableWithLineage) {
+      IcebergCommittable committable =
+          ((CommittableWithLineage<IcebergCommittable>) value).getCommittable();
+
+      if (committable == null || committable.manifest().length == 0) {
+        return;
+      }
+
+      DeltaManifests deltaManifests;
+      WriteResult writeResult;
+      try {
+        deltaManifests =
+            SimpleVersionedSerialization.readVersionAndDeSerialize(
+                DeltaManifestsSerializer.INSTANCE, committable.manifest());
+        writeResult = FlinkManifestUtil.readCompletedFiles(deltaManifests, io, specs);
+      } catch (Exception e) {
+        LOG.warn(
+            "Miss deltaManifests for table {} at checkpoint {}",
+            tableName,
+            committable.checkpointId(),
+            e);
+        return;
+      }
+
+      TableChange tableChange =
+          new TableChange(
+              Arrays.asList(writeResult.dataFiles()), Arrays.asList(writeResult.deleteFiles()));
+      out.collect(tableChange);
+      FlinkManifestUtil.deleteCommittedManifests(
+          tableName, io, deltaManifests.manifests(), flinkJobId, committable.checkpointId());
+    } else {
+      LOG.warn("Unsupported type of committable message: {}", value.getClass());
```

Comment:
Do we get `CommittableSummary` in the PostCommitFlow?

Reply:
Could we just skip it without a warn log? This is business as usual, and should not generate a warn log